bluesky/plans.py

```diff
@@ -92,9 +92,12 @@ def _pre(self):
         # itself as its argument and returns a generator of messages
         if self.pre_run is not None:
             yield from self.pre_run(self)
+        staged = []  # to track duplicates
```

Comment:
Ah. We were talking past each other. My comment about set was referring the possibility of using `self._objects.values()` -> `set(self._objects.values())` and calling it a day, not the possibility of making `staged` a set.